src/utils/utils_alock.h

```diff
@@ -20,29 +20,16 @@ struct ocf_alock;
 /* async request cacheline lock acquisition callback */
 typedef void (*ocf_req_async_lock_cb)(struct ocf_request *req);
 
-typedef bool (*ocf_cl_lock_line_needs_lock_cb)(struct ocf_alock *alock,
-		struct ocf_request *req, unsigned index);
-
-typedef bool (*ocf_cl_lock_line_is_acting_cb)(struct ocf_alock *alock,
-		struct ocf_request *req, unsigned index);
-
-typedef bool (*ocf_cl_lock_line_is_locked_cb)(struct ocf_alock *alock,
-		struct ocf_request *req, unsigned index, int rw);
+typedef int (*ocf_cl_lock_fast)(struct ocf_alock *alock,
+		struct ocf_request *req, int rw);
 
-typedef void (*ocf_cl_lock_line_mark_locked_cb)(struct ocf_alock *alock,
-		struct ocf_request *req, unsigned index, int rw, bool locked);
-
-typedef ocf_cache_line_t (*ocf_cl_lock_line_get_entry_cb)(
-		struct ocf_alock *alock, struct ocf_request *req,
-		unsigned index);
+typedef int (*ocf_cl_lock_slow)(struct ocf_alock *alock,
+		struct ocf_request *req, int rw, ocf_req_async_lock_cb cmpl);
 
 struct ocf_alock_lock_cbs
 {
-	ocf_cl_lock_line_needs_lock_cb line_needs_lock;
-	ocf_cl_lock_line_is_acting_cb line_is_acting;
-	ocf_cl_lock_line_is_locked_cb line_is_locked;
-	ocf_cl_lock_line_mark_locked_cb line_mark_locked;
-	ocf_cl_lock_line_get_entry_cb line_get_entry;
+	ocf_cl_lock_fast lock_lines_fast;
+	ocf_cl_lock_slow lock_lines_slow;
```

Comment:
I think that `lock_entries_*` would be better, as the locked entries are not always cache lines. This also applies to naming convention in entire mio concurrency.